modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/nbt/NbtPathRenderer.kt

```
@@ -0,0 +1,59 @@
+                '"' -> append("\\\"")
+                '\b' -> append("\\b")
+                '\t' -> append("\\t")
+                '\n' -> append("\\n")
+                '\r' -> append("\\r")
+                else -> if (ch.code < 0x20) append("\\u%04x".format(ch.code)) else append(ch)
+            }
+        }
+    }
+
+private fun renderKey(key: String): String =
+    if (key.all { it.isLetterOrDigit() || it == '_' || it == '+' || it == '-' }) {
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)